### DISCUSSION TITLE
some ideas for livewire

### DISCUSSION THREAD

**Discussion by sajjadef98** [livewire/livewire#5061] [3. Feature ideas]
1- livewire for codeigniter
2- livewire native like react netive 
3- livewire pc app builder exe or any like that

**Comment by josegus** [comment]
#2, do you mean livewire for creating mobile apps?

**Reply by rw4lll** [reply]
I think #1 is impossible, because livewire is built specially for Laravel ecosystem. It has no abstract layer to connect it with other frameworks or use standalone. At least, until the next major version. And event if they do that, I don't think they will do official codeigniter support as this framework is not so popular now.
#2 and #3 are also impossible, as it is frontend + backend solution, not just a frontend. For #2 and #3 you can use solutions for such js frameworks as Vue/React.

**Comment by mauricekindermann** [comment]
I thought I was the first person to make this connection, but I'm years late. 

If it ever got built, this would likely be a big inspiration: https://turbo.hotwired.dev/handbook/native (or React native, but that doesn't look super fun to use)

"Impossible" is strong... but I agree livewire itself might not be the best vehicle. But I can imagine livewire style syntax, or an adjacent library, could be made to allow you to create PHP driven mobile app inside of Laravel, and compiled into a binary you could test. 

However given the current focus and interests of Laravel/Caleb, I doubt this would come from there. 

It would be fun experiement to figure out the smallest possible binary you could generate, then slowly layer logic. Just imagine 🤤...

```php
class CounterComponent extends MobileComponent
{
    public $count = 0;
    
    public function increment()
    {
        $this->count++;
        $this->native->haptic('light');
    }
    
    public function template()
    {
        return <<<'mobile'
        <m-screen>
            <m-stack align="center" spacing="4">
                <m-text>Current count: {{ $count }}</m-text>
                <m-button wire:tap="increment">+1</m-button>
            </m-stack>
        </m-screen>
        mobile;
    }
}
```
